ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/AttestationAggregatorTest.java

```diff
@@ -0,0 +1,166 @@
+  private final ChainStorageClient storageClient = new ChainStorageClient(mock(EventBus.class));
+  private AttestationGenerator attestationGenerator = new AttestationGenerator(validatorKeys);
+  private AttestationAggregator aggregator;
+
+  @BeforeEach
+  void setup() {
+    BeaconChainUtil.initializeStorage(storageClient, validatorKeys);
+    aggregator = new AttestationAggregator();
+  }
+
+  @Test
+  void addOwnValidatorAttestation_newData() throws Exception {
+    Attestation attestation = attestationGenerator.validAttestation(storageClient);
+    int validatorIndex = new Random().nextInt(1000);
+    aggregator.committeeIndexToAggregatorInformation.put(
+        attestation.getData().getIndex(),
+        new AggregatorInformation(BLSSignature.random(), validatorIndex));
+    aggregator.addOwnValidatorAttestation(attestation);
+    Attestation aggregateAttesation = aggregator.getAggregateAndProofs().get(0).getAggregate();
+    assertEquals(attestation, aggregateAttesation);
+  }
+
+  @Test
+  void addOwnValidatorAttestation_oldData_noNewAttester() throws Exception {
+    Attestation attestation = attestationGenerator.validAttestation(storageClient);
+    BLSSignature sig = attestation.getAggregate_signature();
+    int validatorIndex = new Random().nextInt(1000);
+    aggregator.committeeIndexToAggregatorInformation.put(
+        attestation.getData().getIndex(),
+        new AggregatorInformation(BLSSignature.random(), validatorIndex));
+    aggregator.addOwnValidatorAttestation(attestation);
+    Attestation newAttestation = new Attestation(attestation);
+    newAttestation.setAggregate_signature(BLSSignature.random());
+    aggregator.addOwnValidatorAttestation(newAttestation);
+    assertEquals(aggregator.getAggregateAndProofs().size(), 1);
+    assertEquals(
+        sig, aggregator.getAggregateAndProofs().get(0).getAggregate().getAggregate_signature());
+  }
+
+  @Test
+  void addOwnValidatorAttestation_oldData_newAttester() throws Exception {
+    Attestation attestation = attestationGenerator.validAttestation(storageClient);
+    BLSSignature sig1 = attestation.getAggregate_signature();
+    int validatorIndex = new Random().nextInt(1000);
+    aggregator.committeeIndexToAggregatorInformation.put(
+        attestation.getData().getIndex(),
+        new AggregatorInformation(BLSSignature.random(), validatorIndex));
+    aggregator.addOwnValidatorAttestation(attestation);
+    Attestation newAttestation = AttestationGenerator.withNewSingleAttesterBit(attestation);
+    BLSSignature sig2 = BLSSignature.random();
+    newAttestation.setAggregate_signature(sig2);
+    aggregator.addOwnValidatorAttestation(newAttestation);
+    assertEquals(aggregator.getAggregateAndProofs().size(), 1);
+    assertEquals(
+        aggregator.getAggregateAndProofs().get(0).getAggregate().getAggregate_signature(),
+        BLSAggregate.bls_aggregate_signatures(List.of(sig1, sig2)));
```

Comment:
Should we also be checking the aggregation bits are correct (here and in the other tests)?

Reply:
done.